packages/restify/lib/restify_mw.js

```diff
@@ -85,7 +68,7 @@ var restifyMW = {
 
     server.on('after', function handledError(req, res, route, err) {
       if (segment && err) {
-        segment.addError(err);
+        segment.close(err);
```

Comment:
This will also make the `closeSegment()` API in `express_mw` a bit of a misnomer, so a quick clarifying comment explaining this fix would also be nice. We can't change the API itself without a major version bump.

Reply:
I've rolled this change back in the `restify` middleware.

I'm not really comfortable rolling a behaviour change for the Express middleware in with this PR though — I think that belongs in another PR really. Also, I think we'd benefit from some proper integration tests with Express to ensure that the middleware is behaving properly.

Reply:
That's totally fair re: express changes. Can save it for a future PR.

We actually do have some integration tests in the repo [here](https://github.com/aws/aws-xray-sdk-node/tree/master/packages/test_express) that have been largely abandoned, but this is a fair reminder that we should really re-incorporate them into the workflow